zerver/lib/string_validation.py

```diff
@@ -0,0 +1,37 @@
+import unicodedata
+
+from django.utils.translation import gettext as _
+
+from zerver.lib.exceptions import JsonableError
+from zerver.models import Stream
+
+unicode_non_chars = [
+    chr(x)
+    for x in list(range(0xFDD0, 0xFDF0))  # FDD0 through FDEF, inclusive
+    + list(range(0xFFFE, 0x110000, 0x10000))  # 0xFFFE, 0x1FFFE, ... 0x10FFFE inclusive
+    + list(range(0xFFFF, 0x110000, 0x10000))  # 0xFFFF, 0x1FFFF, ... 0x10FFFF inclusive
+]
+
+
+def check_invalid_unicode_character(var: str) -> int:
+    for i in range(len(var)):
+        unicode_character = unicodedata.category(var[i])
+        if (unicode_character in ["Cc", "Cs"]) or var[i] in unicode_non_chars:
+            return i
+    return -1
```

Comment:
I'm not a big fan of -1 as an error value, because it's easy to forget to check for.  Making this be typed `Optional[int]` and return `None` would mean mypy would helpfully force us to enforce that we checked for `is not None` before we can do anything with it that assumes it's an int.

Looking at how it's currently called, I guess that wouldn't help here, but I'd prefer to avoid the -1 convention.

Reply:
I tried the same earlier but was facing a problem in using `Optional`, it is now resolved.